server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java

```diff
@@ -31,280 +31,255 @@
+import static org.sonar.db.component.ComponentTesting.newFileDto;
+import static org.sonar.db.component.ComponentTesting.newProjectDto;
+import static org.sonar.db.protobuf.DbFileSources.Test.TestStatus.OK;
+import static org.sonar.server.test.db.TestTesting.newTest;
+import static org.sonar.server.test.ws.ListAction.SOURCE_FILE_ID;
+import static org.sonar.server.test.ws.ListAction.SOURCE_FILE_KEY;
+import static org.sonar.server.test.ws.ListAction.SOURCE_FILE_LINE_NUMBER;
+import static org.sonar.server.test.ws.ListAction.TEST_FILE_ID;
+import static org.sonar.server.test.ws.ListAction.TEST_FILE_KEY;
+import static org.sonar.server.test.ws.ListAction.TEST_ID;
 
 public class ListActionTest {
 
-  private static final String DEFAULT_ORGANIZATION_UUID = "org1";
-
-  @Rule
-  public EsTester es = new EsTester(new TestIndexDefinition(new MapSettings()));
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
   @Rule
+  public EsTester es = new EsTester(new TestIndexDefinition(new MapSettings()));
+  @Rule
   public UserSessionRule userSessionRule = UserSessionRule.standalone();
   @Rule
-  public DbTester db = DbTester.create(System2.INSTANCE)
-      .setDefaultOrganizationUuid(DEFAULT_ORGANIZATION_UUID);
+  public DbTester db = DbTester.create(System2.INSTANCE);
 
   private DbClient dbClient = db.getDbClient();
 
-  private WsTester ws;
+  private TestIndex testIndex = new TestIndex(es.client());
+  private TestIndexer testIndexer = new TestIndexer(System2.INSTANCE, db.getDbClient(), es.client());
+
+  private ComponentDto project;
+  private ComponentDto mainFile;
+  private ComponentDto testFile;
+
+  private WsActionTester ws = new WsActionTester(new ListAction(dbClient, testIndex, userSessionRule, new ComponentFinder(dbClient)));
 
   @Before
-  public void setUp() {
-    TestIndex testIndex = new TestIndex(es.client());
-    ws = new WsTester(new TestsWs(new ListAction(dbClient, testIndex, userSessionRule, new ComponentFinder(dbClient))));
+  public void setUp() throws Exception {
+    project = db.components().insertComponent(newProjectDto(db.getDefaultOrganization()));
+    mainFile = db.components().insertComponent(newFileDto(project));
+    testFile = db.components().insertComponent(newFileDto(project).setQualifier(UNIT_TEST_FILE));
   }
 
   @Test
-  public void list_based_on_test_uuid() throws Exception {
-    userSessionRule.addProjectUuidPermissions(UserRole.CODEVIEWER, TestFile1.PROJECT_UUID);
-
-    dbClient.componentDao().insert(db.getSession(), TestFile1.dto());
-    db.getSession().commit();
+  public void return_test() throws Exception {
+    userSessionRule.addProjectUuidPermissions(CODEVIEWER, project.uuid());
+    DbFileSources.Test test = newTest(mainFile, 10, 11, 12, 20, 21, 25).setStatus(OK).build();
+    insertTests(testFile, test);
+
+    ListResponse request = call(ws.newRequest().setParam(TEST_ID, test.getUuid()));
+
+    assertThat(request.getTestsList()).hasSize(1);
```

Comment:
comment: I think we should make a better use of custom assertJ assertion, at least we should try